gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/finder/DatasetFinder.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention.dataset.finder;
+
+import java.io.IOException;
+import java.util.List;
+
+import gobblin.data.management.retention.dataset.Dataset;
+
+
+/**
+ * Finds {@link gobblin.data.management.retention.dataset.Dataset}s in the file system.
+ *
+ * <p>
+ *   Concrete subclasses should have a constructor with signature
+ *   ({@link org.apache.hadoop.fs.FileSystem}, {@link azkaban.utils.Props}).
+ * </p>
+ */
+public interface DatasetFinder {
+
+  /**
+   * Find all {@link gobblin.data.management.retention.dataset.Dataset}s in the file system.
+   * @return List of {@link gobblin.data.management.retention.dataset.Dataset}s in the file system.
+   * @throws IOException
+   */
+  public List<Dataset> findDatasets() throws IOException;
```

Comment:
Actually, after thinking about it, I would prefer if DatasetFinder handled its own filesystems. 
The way I see it, if wanting datasets for multipe users, the dataset finder would itself switch filesystems based on the user needed (in this case, the fs passed at constructor would be used only to get the URI of the file system).
The problem of the method `findDatasets(FileSystem fs)` is that this makes the driver program responsible for figuring out file systems, and I would prefer this responsibility to be delegated to the "MultiUserDatasetFinder". Any thoughts?